lib/Sema/TypeCheckPropertyWrapper.cpp

```diff
@@ -174,16 +174,31 @@ static ConstructorDecl *findInitialValueInit(ASTContext &ctx,
 static ConstructorDecl *findDefaultInit(ASTContext &ctx,
                                         NominalTypeDecl *nominal) {
   SmallVector<ConstructorDecl *, 2> defaultValueInitializers;
-  DeclName initName(ctx, DeclBaseName::createConstructor(),
-                    ArrayRef<Identifier>());
   SmallVector<ValueDecl *, 2> decls;
+  auto initName = DeclBaseName::createConstructor();
   nominal->lookupQualified(nominal, initName, NL_QualifiedDefault, decls);
   for (const auto &decl : decls) {
     auto init = dyn_cast<ConstructorDecl>(decl);
     if (!init || init->getDeclContext() != nominal)
       continue;
 
-    defaultValueInitializers.push_back(init);
+    // A constructor which does not have any parameters or where all the
+    // parameters have a default argument can be used to default initialize
+    // the property wrapper type.
+    assert(init->hasParameterList());
+    auto hasParams = init->getParameters()->size() > 0;
+    auto allParamsHaveDefaultArg = false;
+
+    if (hasParams) {
```

Comment:
I don't think we need to check `hasParams`? `allParametersHaveDefaultArg` is trivially `true` if there are 0 parameters.

There might be a failure case here where the initializer has generic parameters that won't get inferred if you don't pass a parameter, which means it won't be able to be called. For example `init<T: ExpressibleByIntegerLiteral>(_ value: T = 0)`

Reply:
Ah, you’re right. I’ll remove the check. I suppose we can skip an init() where it has generic parameters (and default arguments).

Reply:
Thanks!